File: backproduit/src/main/java/tn/esprit/examen/nomPrenomClasseExamen/entities/Produit.java
package tn.esprit.examen.nomPrenomClasseExamen.entities;

import jakarta.persistence.*;
import lombok.*;



@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Produit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;
    private String description;
    private double prix;
    private int stock;
    private String imageUrl; // Stocke l'URL de l'image

    @ManyToOne

    private CategorieProduit categorie;
}

File: frontproduit/src/app/models/produit.ts
import { CategorieProduit } from "./categorie";

export class Produit {
    id?: number;
    nom: string;
    description: string;
    prix: number;
    stock: number;
    imageUrl: string;
    categorie: CategorieProduit;
  
    constructor(
      nom: string,
      description: string,
      prix: number,
      stock: number,
      imageUrl: string,
      categorie: CategorieProduit,
      id?: number
    ) {
      this.nom = nom;
      this.description = description;
      this.prix = prix;
      this.stock = stock;
      this.imageUrl = imageUrl;
      this.categorie = categorie;
      this.id = id;
    }
  }
  